### DISCUSSION TITLE
Why does my GMap build show a blank screen?

### DISCUSSION THREAD

**Discussion by claww** [BimmerGestalt/AAIdrive#225] [Q&A] [answered]
Hi Hufman,

I build the apk for dirty-gmap and it show the icon in IDrive but actually when I open the map, its a blank screen, it shows google in the left down side as on google maps, but actually there are no maps. I have done anything wrong ? 
I have idrive5, apps connect via Bluetooth.

Thanks
Claudius

**Comment by hufman** [comment, marked as answer]
This is usually because the key didn't get included into the build properly, or the key doesn't have the correct permissions in the [Google Cloud Console](https://console.cloud.google.com/google/maps-apis/credentials).

In Android Studio, use the Build > Analyze APK tool to open your output APK's AndroidManifest.xml, and you should see your key in the `<meta-data android:name="com.google.android.geo.API_KEY" android:value="{KEY}">` location. If not, Gradle didn't pick up the key properly from gradle.properties.

If that does show the correct key, then the other possibility is that the key doesn't have the correct API permissions. The definitive proof of this will be `adb logcat` output showing errors, but they are difficult to find among all the other log messages.

The Credential Key App Restrictions are not necessary, and I have my personal build set to None.
The Credential Key API Restrictions in the Google Cloud Console should look like this:
![image](https://user-images.githubusercontent.com/1592375/109749990-7c8c2b00-7b90-11eb-8f74-acdc0c6d6bd0.png)

**Reply by claww** [reply]
Thank you for detailed answer, I didn't had enabled in API key those 3 APIs. However now I'm able to see the map but I'm not able to search for a place to navigate.
When I do search for a place, doesn't matter what I'm writing there, it say 'no entries'.


API | Requests
-- | --
Places API | 64
Maps SDK for Android | 3
Directions API | 0



API | Requests | Errors | Avg latency (ms) |  
-- | -- | -- | -- | --
Directions API | 0 | 0 | - | Details
Maps SDK for Android | 3 | 0 | - | Details
Places API | 64 | 64 | 14 | Details

An error occurred retrieving ClientID. ( this popup at bottom of cloud console )

**Reply by hufman** [reply]
Excellent, I'm glad that worked, and now it's a Q&A for others to use :)
As for the Places, it seems that you also have encountered #180, which I haven't been able to figure out yet. Do you have billing enabled, maybe Places requires Billing now? I haven't been able to figure out what is different from my key to others...

**Reply by claww** [reply]
Forgot to mention that I do not have billing enabled, its safe to enable, do they provide a limited requests per month ?

**Reply by hufman** [reply]
Yes they provide a [$200/mo credit](https://cloud.google.com/maps-platform/pricing) which should be more than enough to handle your own usage.

**Reply by claww** [reply]
After enable billing, it works! Thanks a lot for your support!
